src/template.js

```diff
@@ -87,14 +87,13 @@ export const TemplateChunk = Object.freeze({
     // create the DOM if it wasn't created before
     this.createDOM(el)
 
-    if (this.dom) {
-      // create the new template dom fragment if it want already passed in via meta
-      this.fragment = fragment || this.dom.cloneNode(true)
-    }
+    // create the new template dom fragment if it want already passed in via meta
+    this.fragment = fragment || this.dom.cloneNode(true)
```

Comment:
When I debugged this code I got a question: Why we need to use `fp/no-let` eslint rule?

I think this rule is pretty meaningless because variables cannot mutate their values when reassigned. 
I like the rule more https://eslint.org/docs/rules/prefer-const

Maybe this line is not the best example of what I want to explain, but it is pretty close. All that I will write further does not concern this code, just abstract examples.

It seems quite confusing to me that there are two variables: `this.template` and just `template`, which can have the same value, or they can differ.  Since `this.template` is more up-to-date, the code will refer to it later, but this leads to worse minification.

In addition, sometimes you have to create variables in the spirit of `templateOrigin` and work with three at once: `template`, `templateOrigin`, `this.template`.
I think it would be more elegant to work with just `template` in the process and assign it to this at the very end.

For example:

_no-let_
```
const { fragment, children } = meta;
...
// It is rather confusing that fragment and this.fragment are different in some situations
this.fragment = fragment || this.dom.cloneNode(true)
...
// It is rather confusing that children and this.children are different in some situations
this.children = isTemplateTag ? children || Array.from(this.fragment.childNodes) : null
...
// this.fragment is deoptimized for minification
if (!avoidDOMInjection && this.fragment) injectDOM(el, this.fragment)
...
// store the template meta properties    
this.meta = meta
...
```

_prefer-const_
```
let { fragment } = meta;
...
// pseudocode for demo purposes
if (isTemplateTag) {
  fragment = getTemplateFragment(...)
}

if (!fragment) {
  fragment = this.dom.cloneNode(true) 
}

// This is certainly a bad example because in real code it is better to do here const fragment = meta.fragment ?? this.dom.cloneDone(true)
...
const children = isTemplateTag ? meta.children || Array.from(fragment.childNodes) : null
...
if (!avoidDOMInjection && fragment) injectDOM(el, fragment)
...
// store the template properties    
this.meta = meta
this.fragment = fragment
this.children = children
...
```

Reply:
thank you for your feedback, I never use `let` or mutable variables in any of the RIot.js scripts (nor in my private project). I prefer a functional approach avoiding the use of `let` and `var` at all. This is not the right place to discuss about it, however I could manage to improve a bit the semantic of the mount method. I think we are ready for a new patch release. Thank you for your help.